src/main/fc/rc.c

```diff
@@ -341,71 +341,65 @@ static FAST_CODE_NOINLINE void rcSmoothingSetFilterCutoffs(rcSmoothingFilter_t *
-                } else {
-                    pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[i], pt3FilterGain(throttleCutoffFrequency, dT));
-                }
-            }
+    const float setpointCutoffFrequency = smoothingData->setpointCutoffFrequency;
+    const float throttleCutoffFrequency = smoothingData->throttleCutoffFrequency;
+
+    if (!smoothingData->filterInitialized) {
+        for (int i = FD_ROLL; i <= FD_YAW; i++) {
+            pt3FilterInit(&smoothingData->filterSetpoint[i], pt3FilterGain(setpointCutoffFrequency, dT));
+            pt3FilterInit(&smoothingData->filterFeedforward[i], pt3FilterGain(setpointCutoffFrequency, dT));
         }
-        // initialize or update the RC Deflection filter
         for (int i = FD_ROLL; i < FD_YAW; i++) {
-            if (!smoothingData->filterInitialized) {
-                pt3FilterInit(&smoothingData->filterRcDeflection[i], pt3FilterGain(setpointCutoffFrequency, dT));
-            } else {
-                pt3FilterUpdateCutoff(&smoothingData->filterRcDeflection[i], pt3FilterGain(setpointCutoffFrequency, dT));
-            }
+           pt3FilterInit(&smoothingData->filterRcDeflection[i], pt3FilterGain(setpointCutoffFrequency, dT));
         }
-    }
-    // initialize or update the Feedforward filter
-    if ((smoothingData->feedforwardCutoffFrequency != oldFeedforwardCutoff) || !smoothingData->filterInitialized) {
-       for (int i = FD_ROLL; i <= FD_YAW; i++) {
-            const float feedforwardCutoffFrequency = smoothingData->feedforwardCutoffFrequency;
-            if (!smoothingData->filterInitialized) {
-                pt3FilterInit(&smoothingData->filterFeedforward[i], pt3FilterGain(feedforwardCutoffFrequency, dT));
-            } else {
-                pt3FilterUpdateCutoff(&smoothingData->filterFeedforward[i], pt3FilterGain(feedforwardCutoffFrequency, dT));
-            }
+        pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[3], pt3FilterGain(throttleCutoffFrequency, dT));
+    }
+
+    // note that cutoff frequencies are integers, filter cutoffs won't re-calculate until there is > 1hz variation from previous cutoff
+    if (smoothingData->setpointCutoffFrequency != oldSetpointCutoff) {
+        // Update the RC Setpoint/Deflection filter and FeedForward Filter
+        // all cutoffs will be the same, we can optimize :)
+        pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[0], pt3FilterGain(setpointCutoffFrequency, dT));
+        const float pt3K = smoothingData->filterSetpoint[0].k;
+        for (int i = FD_ROLL; i <= FD_YAW; i++) {
+            pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[i], pt3K);
+            pt3FilterUpdateCutoff(&smoothingData->filterFeedforward[i], pt3K);
+        }
+        for (int i = FD_ROLL; i < FD_YAW; i++) {
+            pt3FilterUpdateCutoff(&smoothingData->filterRcDeflection[i], pt3K);
         }
```

Comment:
```suggestion
        for (int i = 0; i <= FD_YAW; i++) {
            pt3FilterUpdateCutoff(&smoothingData->filterSetpoint[i], pt3K);
            pt3FilterUpdateCutoff(&smoothingData->filterFeedforward[i], pt3K);
            pt3FilterUpdateCutoff(&smoothingData->filterRcDeflection[i], pt3K);
        }
```

Reply:
RC deflection only hold pitch and roll. Its an artsy of size 2. This will index outside of the rc deflection array.